src/object.js

```diff
@@ -146,11 +161,40 @@ ReflectionObject.prototype.onRemove = function onRemove(parent) {
 ReflectionObject.prototype.resolve = function resolve() {
     if (this.resolved)
         return this;
+    if (this instanceof Root || this.parent && this.parent.resolved)
+        this._resolveFeatures();
     if (this.root instanceof Root)
-        this.resolved = true; // only if part of a root
+        this.resolved = true;
     return this;
 };
 
+/**
+ * Resolves child features from parent features
+ * @returns {undefined}
+ */
+ReflectionObject.prototype._resolveFeatures = function _resolveFeatures() {
+    var defaults = {};
+
+    if (this instanceof Root) {
+        if (this.root.getOption("syntax") === "proto2") {
```

Comment:
What happens if there's no syntax or edition specification in the .proto file?  That's valid in OSS and results in proto2 behavior

Reply:
Got it.